File: frontend/src/server/trpc/routers/cache.ts
/**
 * ================================================
 * tRPC ROUTER — Cache
 * ================================================
 */

import { router } from '../init';
import { adminProcedure } from '../middleware';
import testmoService from '../../services/testmo.service';
import logger from '../../services/logger.service';

export const cacheRouter = router({
  clear: adminProcedure.mutation(() => {
    testmoService.clearCache();
    logger.info('Cache cleared manually via tRPC');
    return { success: true as const, message: 'Cache cleared successfully', timestamp: new Date().toISOString() };
  }),
});

File: backend_py/app/routers/trpc/cache.py
from typing import Any

from app.services.testmo import testmo_service

from app.routers.trpc._common import _result


async def _cache_clear(_input_data: dict[str, Any] | None, db) -> dict[str, Any]:
    testmo_service.clear_cache()
    return _result({"success": True, "message": "Cache cleared successfully"})
